src/unix.rs

```diff
@@ -348,6 +348,18 @@ pub(crate) fn devicename_os() -> OsString {
     fancy_fallback_os(computer)
 }
 
+#[cfg(target_os = "illumos")]
+pub(crate) fn devicename() -> String {
+    let mut nodename = String::new();
+
+    if let Ok(program) = std::fs::read_to_string("/etc/nodename") {
+        let program = program.into_bytes();
+        nodename.push_str(&String::from_utf8_lossy(&program));
+        nodename.pop();
```

Comment:
Can you add a comment saying which character is being popped here?  Is it a newline, null byte, etc.?

Reply:
Updated!